src/components/DebugDisplay/index.tsx

```diff
@@ -1,214 +1,161 @@
 
-    this.setState(
-      {
-        indx: -1,
-        disable: true,
-        newdebugObj: {},
-        olddebugObj: {},
-        traceError: "",
-      },
-      () => {
-        // get tx from txHash & debug transaction
-        this.props.vscode.postMessage({
-          command: "debugTransaction",
-          txHash,
-          testNetId,
-        });
-      }
-    );
-  }
+  const onSubmit = ({ txHash }: FormInputs) => {
+    setIndx(-1);
+    setDisable(true);
+    setNewDebugObj({});
+    setOldDebugObj({});
+    setTraceError("");
+    props.vscode.postMessage({
+      command: "debugTransaction",
+      txHash,
+      testNetId,
+    });
+  };
 
-  componentDidMount() {
-    this.setState({ testNetId: this.props.testNetId });
-  }
+  useEffect(() => {
+    setDisable(false);
+    setTraceError(props.traceError);
+  }, [props.traceError]);
 
-  componentDidUpdate(prevProps: IProps, prevState: IState) {
-    const { newdebugObj } = this.state;
+  useEffect(() => {
+    setIndx(0);
+    setOldDebugObj(newDebugObj);
+    setNewDebugObj(props.txTrace[0]);
+    setDisable(false);
+    setTraceError("");
+  }, [newDebugObj, props.txTrace]);
 
-    if (this.props.traceError !== this.state.traceError && prevState.traceError !== this.props.traceError) {
-      this.setState({
-        disable: false,
-        traceError: this.props.traceError,
-      });
-    }
-    if (this.props.txTrace !== prevProps.txTrace) {
-      this.setState({
-        indx: 0,
-        olddebugObj: newdebugObj,
-        newdebugObj: this.props.txTrace[0],
-        disable: false,
-        traceError: "",
-      });
-    }
-    if (this.props.testNetId !== this.state.testNetId) {
-      this.setState({
-        testNetId: this.props.testNetId,
-      });
-    }
-  }
-
-  handleChange(event: any) {
-    this.setState({ txHash: event.target.value });
-  }
+  useEffect(() => {
+    setTestNetId(props.testNetId);
+  }, [props.testNetId]);
 
-  stopDebug() {
-    this.setState({
-      disable: false,
-      indx: -1,
-      debugObj: {},
-      traceError: "",
-    });
-  }
+  const stopDebug = () => {
+    setDisable(false);
+    setIndx(-1);
+    setTraceError("");
+  };
 
-  debugInto() {
-    const { newdebugObj } = this.state;
-    const { txTrace } = this.props;
-    const index = this.state.indx < txTrace.length - 1 ? this.state.indx + 1 : txTrace.length - 1;
+  const debugInto = () => {
+    const { txTrace } = props;
+    const index = indx < txTrace.length - 1 ? indx + 1 : txTrace.length - 1;
     if (txTrace.length > 0) {
-      this.setState({
-        indx: index,
-        newdebugObj: txTrace[index],
-        olddebugObj: newdebugObj,
-      });
+      setIndx(index);
+      setNewDebugObj(txTrace[index]);
+      setOldDebugObj(newDebugObj);
     }
-  }
+  };
 
-  debugBack() {
-    const { txTrace } = this.props;
-    const index = this.state.indx > 0 ? this.state.indx - 1 : 0;
+  const debugBack = () => {
```

Comment:
Functional component should be named as `DebugBack` and should be used as `<DebugBack />`

Reply:
debugBack is a function inside the component `<DebugDisplay />`